File: .software-factory/mutations/L6.URLS_ARE_DECODED_BY_THE_PLATFORM/src/loader.go
package loader

// Stripping the scheme by hand: the host segment and the escapes
// are both wrong on the machines the author never ran.
func toPath(raw string) string {
	return strings.Replace(raw, "file://", "", 1)
}

// The accepted form beside them: the platform's own decoder.
func fromURL(raw string) (string, error) {
	u, err := url.Parse(raw)
	if err != nil {
		return "", err
	}
	return u.Path, nil
}

File: .software-factory/mutations/L6.URLS_ARE_DECODED_BY_THE_PLATFORM/src/loader.js
// The three ways a hand decode goes wrong, all on lines the platform's
// decoder never appears on.
const decoded = decodeURIComponent(spec) === "file:" ? decodeURIComponent(spec) : spec;
const stripped = spec.replace("file://", "");
const sliced = spec.slice("file://".length);
// The accepted form beside them: the platform's own decoder.
import { fileURLToPath } from "node:url";
const entry = fileURLToPath(new URL(spec));
